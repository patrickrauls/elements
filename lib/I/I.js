import React from 'react';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import regular from '@fortawesome/fontawesome-free-regular';
import solid from '@fortawesome/fontawesome-free-solid';
import PropTypes from 'prop-types';

fontawesome.library.add(regular, solid);

const I = ({ className, ...rest }) => {
  return (
    <FontAwesomeIcon
      {...rest}
      className={className}
    />
  );
};

I.propTypes = {
  className: PropTypes.string,
  hidden: PropTypes.bool,
  icon: PropTypes.string.isRequired
};

I.defaultProps = {
  className: '',
  hidden: false
};

export default I;
