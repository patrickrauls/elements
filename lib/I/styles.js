import { css } from 'styled-components';

export default css`
display: ${props => props.hidden ? 'none' : 'inline'};
`;
