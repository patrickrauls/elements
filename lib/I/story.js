import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';
import I from './index';
// import withTheme from '../../../stories/withTheme';

storiesOf('I', module)
  // .addDecorator(withTheme);
  .addDecorator(withKnobs)
  .add('icon', () => (
    <I
      icon={select('icon', ['user', 'lock', 'globe'], 'user')}
    />
  ));