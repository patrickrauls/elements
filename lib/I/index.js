import styled from 'styled-components';

import I from './I';
import styles from './styles';

const StyledI = styled(I)`${styles}`;

export default StyledI;
