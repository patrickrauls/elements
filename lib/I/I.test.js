import React from 'react';
import I from './I';
import StyledI from './index';
import toJson from 'enzyme-to-json'
import 'jest-styled-components';


describe('I (icon component)', () => {
  describe('unit tests', () => {
    it('renders without crashing', () => {
      const wrapper = shallow(<I icon="check" />);
      expect(wrapper.exists()).toBe(true);
    });
  });
  describe('ui tests', () => {
    const setup = props => shallow(<StyledI icon="check" {...props} />);
    it('adds styles to the component', () => {
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
    it('displays the icon when not given a hidden prop', () => {
      const wrapper = setup();
      expect(toJson(wrapper)).not.toHaveStyleRule('display', 'none');
    });
    it('hides the icon when given hidden prop', () => {
      const wrapper = setup({ hidden: true });
      expect(toJson(wrapper)).toHaveStyleRule('display', 'none');
    });
  });
});
