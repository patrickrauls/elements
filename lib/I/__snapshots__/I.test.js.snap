// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I (icon component) ui tests adds styles to the component 1`] = `
.c0 {
  display: inline;
}

<I
  className="c0"
  hidden={false}
  icon="check"
/>
`;
